src/browser/progressbar.js

```diff
@@ -26,72 +62,43 @@ define((require, exports, module) => {
     }),
   });
 
-  const PI = 3.1416; // > PI
-
-  // Animation parameters:
-  const A = 0.2;              // Zone A size (a full progress is equal to '1'
-  const B = 0.3;              // Zone B size
-  const APivot = 200;         // When to reach ~80% of zone A
-  const BPivot = 500;         // When to reach ~80% of zone B
-  const CSpeed = 0.002;       // Speed to fill zone C (how fast we reach '1')
-  const StartFading = 0.8;    // When does opacity starts decreasing to 0
-
-  // Inverse tangent function: [0 - inf] -> [0 - PI/2]
-  // ApproachFunc: [time, pivot] -> [0 - 1]
-  // Pivot value is more or less when the animation seriously starts to slow down
-  const ApproachFunc = (tMs, pivoMs) => 2 * Math.atan(tMs / pivoMs) / PI;
-
-  // rfa and before are global and not associated to any state. They are used
-  // to drive the progressbar animation. There's only one animation to play
-  // at the same time.
-  let rfa = -1;
-  let before = 0;
-
   const ProgressBar = Component([{
-    step(now) {
+    step() {
+      let rfa = requestAnimationFrame(() => this.step());
+      this.props.rfaCursor.set('id', rfa);
+    },
+    componentDidUpdate() {
       const viewer = this.props.webViewerCursor;
-
-      let progress; // value between 0 and 1
-
-      if (viewer.get('isLoading')) {                                            // Zone A
-        const startLoadingTime = viewer.get('startLoadingTime');
-        if (viewer.get('isConnecting')) {
-          progress = A * ApproachFunc(now - startLoadingTime, APivot);
-        } else {                                                                // Zone B
-          const connectedAt = viewer.get('connectedAt');
-          const A_offset = A * ApproachFunc(connectedAt - startLoadingTime, APivot);
-          progress = A_offset + B * ApproachFunc(now - connectedAt, BPivot);
-        }
-      } else {                                                                  // Zone C
-        let lastProgress = viewer.get('progress');
-        if (lastProgress < 1) {
-          progress = lastProgress + (now - before) * CSpeed;
-        } else {
-          progress = 1;
+      // Stop if loaded and had enough time to draw the final animation
+      if (viewer.get('readyState') && !viewer.get('isLoading')) {
```

Comment:
Nit: I'm not sure we may have `isLoading: true` without `readyState` that would seem odd. If we do have such case we should probably